zuul-core/src/main/java/com/netflix/netty/common/proxyprotocol/ElbProxyProtocolChannelHandler.java

```diff
@@ -16,142 +16,43 @@
-                            case TCP4:
-                            case TCP6:
-                                addr = new InetSocketAddress(
-                                        InetAddresses.forString(destinationAddress), hapm.destinationPort());
-                                break out;
-                            case UNIX_STREAM: // TODO: implement
-                            case UDP4:
-                            case UDP6:
-                            case UNIX_DGRAM:
-                                throw new IllegalArgumentException("unknown proxy protocol" + destinationAddress);
-                        }
-                        throw new AssertionError(hapm.proxiedProtocol());
-                    }
-                    channel.attr(SourceAddressChannelHandler.ATTR_LOCAL_ADDR).set(addr);
-                }
-
-                // Get the real client IP from the ProxyProtocol message sent by the ELB, and overwrite the SourceAddress
-                // channel attribute.
-                String sourceAddress = hapm.sourceAddress();
-                if (sourceAddress != null) {
-                    channel.attr(SourceAddressChannelHandler.ATTR_SOURCE_ADDRESS).set(sourceAddress);
-                    channel.attr(SourceAddressChannelHandler.ATTR_SOURCE_PORT).set(hapm.sourcePort());
-
-                    SocketAddress addr;
-                    out: {
-                        switch (hapm.proxiedProtocol()) {
-                            case UNKNOWN:
-                                throw new IllegalArgumentException("unknown proxy protocl" + sourceAddress);
-                            case TCP4:
-                            case TCP6:
-                                addr = new InetSocketAddress(
-                                        InetAddresses.forString(sourceAddress), hapm.sourcePort());
-                                break out;
-                            case UNIX_STREAM: // TODO: implement
-                            case UDP4:
-                            case UDP6:
-                            case UNIX_DGRAM:
-                                throw new IllegalArgumentException("unknown proxy protocol" + sourceAddress);
-                        }
-                        throw new AssertionError(hapm.proxiedProtocol());
-                    }
-                    channel.attr(SourceAddressChannelHandler.ATTR_REMOTE_ADDR).set(addr);
-                }
-
-                // TODO - fire an additional event to notify interested parties that we now know the IP?
-
-                // Remove ourselves (this handler) from the channel now, as no more work to do.
-                ctx.pipeline().remove(this);
-
-                // Do not continue propagating the message.
-                return;
-            }
+    public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+        if (withProxyProtocol && isHAPMDetected(msg)) {
```

Comment:
I thought you mentioned the HAPM decoder can ignore this?

Reply:
It does try to parse the header bytes unconditionally, and blows up in case it's not a HAPM. That is the likely the historical "context" around the optional wiring of the decoder.